packages/widget/src/components/common/dropdown/dropdown.ts

```diff
@@ -0,0 +1,145 @@
+import { html, type HTMLTemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { map } from 'lit/directives/map.js';
+import { when } from 'lit/directives/when.js';
+import { chevronIcon, networkIconsMap } from '../../../assets';
+import { capitalize } from '../../../utils';
+import { BaseComponent } from '../../base-component/base-component';
+import { styles } from './styles';
+
+export interface DropdownOption {
+  id?: string;
+  name: string;
+  icon?: HTMLTemplateResult | string;
+  value?: string | object;
+}
+
+@customElement('dropdown-component')
+export class Dropdown extends BaseComponent {
+  static styles = styles;
+
+  @state()
+  isDropdownOpen = false;
+
+  @property({ type: Boolean })
+  disabled = false;
+
+  @property({ type: String })
+  placeholder = '';
+
+  @property({ type: String })
+  label = '';
+
+  @property({ type: Array })
+  options: DropdownOption[] = [];
+
+  @property({ type: Object })
+  selectedOption: DropdownOption | null = null;
+
+  @property({ attribute: false })
+  onOptionSelected: (option?: DropdownOption) => void = () => {};
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    addEventListener('click', this._handleOutsideClick);
+  }
+
+  disconnectedCallback(): void {
+    super.connectedCallback();
+    removeEventListener('click', this._handleOutsideClick);
+  }
+
+  _handleOutsideClick = (event: MouseEvent): void => {
+    if (this.isDropdownOpen && !event.composedPath().includes(this)) {
+      document.removeEventListener('click', this._handleOutsideClick);
+      this.isDropdownOpen = false;
+    }
+  };
+
+  _toggleDropdown = (): void => {
+    if (!this.disabled) {
+      this.isDropdownOpen = !this.isDropdownOpen;
+    }
+  };
+
+  _selectOption(option: DropdownOption, event: Event): void {
+    if (this.disabled) return;
+
+    event.stopPropagation();
+    this.selectedOption = option;
+    this._toggleDropdown();
+    this.dispatchEvent(
+      new CustomEvent('option-selected', { detail: { option } })
```

Comment:
you can just emit `value` instead of whole option, so there won't be a need for unwrapping in handler function